.github/workflows/ci.yaml

```diff
@@ -0,0 +1,99 @@
+# yamllint disable rule:line-length
+
+name: Continuous Integration
+
+on:
+  pull_request:
+  push:
+    branches:
+      - '[0-9].[0-9]+'
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        operating-system:
+          - ubuntu-latest
+        php-version:
+          - 7.1
+          - 7.2
+          - 7.3
+          - 7.4
+
+        include:
+          - operating-system: 'ubuntu-latest'
+            php-version: 8.0
+            composer-flags: '--ignore-platform-req=php'
+            PHP_CS_FIXER_IGNORE_ENV: 1
+
+          - operating-system: 'ubuntu-latest'
+            php-version: 5.6
+            PHP_CS_FIXER_TEST_USE_LEGACY_TOKENIZER: 1
+            SYMFONY_DEPRECATIONS_HELPER: disabled
+
+          - operating-system: 'windows-latest'
+            php-version: 7.3
+            FAST_LINT_TEST_CASES: 1
+
+          - operating-system: 'windows-latest'
+            php-version: 5.6
+            SKIP_LINT_TEST_CASES: 1
+
+          - operating-system: 'macos-latest'
+            php-version: 7.4
+            PHP_CS_FIXER_FUTURE_MODE: 1
+
+    name: PHP ${{ matrix.php-version }} CI on ${{ matrix.operating-system }} ${{ matrix.composer-flags }}
+
+    runs-on: ${{ matrix.operating-system }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-dir)"
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('composer.*') }}-${{ matrix.composer-flags }}
+          restore-keys: |
+            composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('composer.*') }}-
+            composer-${{ runner.os }}-${{ matrix.php-version }}-
+            composer-${{ runner.os }}-
+            composer-
+
+      - name: Install dependencies
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          retry_wait_seconds: 30
+          command: |
+            composer update --optimize-autoloader --no-interaction --no-progress --ignore-platform-req=hhvm ${{ matrix.composer-flags }}
+
+      - name: Run tests
+        continue-on-error: ${{ matrix.php-version == '8.0' }}
```

Comment:
Several tests are broken on PHP 8 since not very long time ago due to a bug in TimeEfficientLongestCommonSubsequenceImplementation. It looks like this:

```1) PhpCsFixer\Tests\Differ\SebastianBergmannShortDifferTest::testDiffReturnsDiff
Undefined array key 140725147700175

/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/vendor/php-cs-fixer/diff/src/v1_4/LCS/TimeEfficientLongestCommonSubsequenceImplementation.php:57
/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/vendor/php-cs-fixer/diff/src/v1_4/Differ.php:230
/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/vendor/php-cs-fixer/diff/src/v1_4/Differ.php:55
/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/src/Differ/SebastianBergmannShortDiffer.php:37
/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/tests/Differ/SebastianBergmannShortDifferTest.php:36
/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer/vendor/phpunitgoodpractices/traits/src/ExpectationViaCodeOverAnnotationTrait7.php:39
```

Reply:
This might be a PHP 8 bug FWIW.